### PR TITLE
[5.1] Upgrade to guzzle 6

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

**Comment by mtdowling** [comment]
:shipit:

**Comment by taylorotwell** [comment]
Where we were using PostFile before, how does it know to use a file since we're just passing a string?

**Comment by mtdowling** [comment]
Good catch. If the filename is required, then you'll need to add one by passing in the "filename" attribute to the hash. I wasn't sure if it was necessary or not. Alternatively, you can pass the contents in as an `fopen` resource and the filename will be extracted automatically.

**Comment by GrahamCampbell** [comment]
@mtdowling Just been testing out the mailgun transport, and it doesn't actually work.

Raw stack trace from my app:

```
GuzzleHttp\Exception\ClientException Client error: 400
    vendor/guzzlehttp/guzzle/src/Middleware.php:68 GuzzleHttp\{closure}
    vendor/guzzlehttp/promises/src/Promise.php:199 callHandler
    vendor/guzzlehttp/promises/src/Promise.php:165 GuzzleHttp\Promise\{closure}
    vendor/guzzlehttp/promises/src/FulfilledPromise.php:39 GuzzleHttp\Promise\{closure}
    vendor/guzzlehttp/promises/src/TaskQueue.php:60 run
    vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php:96 tick
    vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php:123 execute
    vendor/guzzlehttp/promises/src/Promise.php:240 invokeWaitFn
    vendor/guzzlehttp/promises/src/Promise.php:217 waitIfPending
    vendor/guzzlehttp/promises/src/Promise.php:261 invokeWaitList
    vendor/guzzlehttp/promises/src/Promise.php:219 waitIfPending
    vendor/guzzlehttp/promises/src/Promise.php:62 wait
    vendor/guzzlehttp/guzzle/src/Client.php:130 request
    vendor/guzzlehttp/guzzle/src/Client.php:88 __call
    vendor/laravel/framework/src/Illuminate/Mail/Transport/MailgunTransport.php:87 post
    vendor/laravel/framework/src/Illuminate/Mail/Transport/MailgunTransport.php:87 send
    vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php:85 send
    vendor/laravel/framework/src/Illuminate/Mail/Mailer.php:366 sendSwiftMessage
    vendor/laravel/framework/src/Illuminate/Mail/Mailer.php:168 send
    app/Handlers/Events/AnalysisNotificationsHandler.php:114 sendMessages
    app/Handlers/Events/AnalysisNotificationsHandler.php:77 handle
    [internal]:0 call_user_func_array
    vendor/laravel/framework/src/Illuminate/Events/CallQueuedHandler.php:41 call
    vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php:126 resolveAndFire
    vendor/laravel/framework/src/Illuminate/Queue/Jobs/BeanstalkdJob.php:51 fire
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:207 process
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:159 pop
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:113 runNextJobForDaemon
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:85 daemon
    vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php:101 runWorker
    vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php:67 fire
    [internal]:0 call_user_func_array
    bootstrap/cache/compiled.php:958 call
    vendor/laravel/framework/src/Illuminate/Console/Command.php:150 execute
    vendor/symfony/console/Command/Command.php:259 run
    vendor/laravel/framework/src/Illuminate/Console/Command.php:136 run
    vendor/symfony/console/Application.php:878 doRunCommand
    vendor/symfony/console/Application.php:195 doRun
    vendor/symfony/console/Application.php:126 run
    vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:94 handle
    artisan:45 [main]
```

Sorry, don't have time to collect more info. Got to sleep.

**Comment by mtdowling** [comment]
We'd need to see the actual HTTP response message to know what the error is. It's possible that you need the "filename" attribute. See https://github.com/laravel/framework/pull/8907#issuecomment-106067408

**Comment by GrahamCampbell** [comment]
Here's some more info then:

image

**Comment by GrahamCampbell** [comment]
Right, this stuff works if we include the "filename". :)

**Comment by GrahamCampbell** [comment]
Tested with both mandrill and mailgun. Both working. :)

**Comment by jeremeamia** [comment]
Cool!

**Comment by lucasmichot** [comment]
Cool, but only if you plan to upgrade your PHP version : https://github.com/guzzle/guzzle/blob/master/composer.json#L16
cc @GrahamCampbell @taylorotwell

**Comment by GrahamCampbell** [comment]
@lucasmichot Yeh, we're already looking into supporting guzzle 5 too. :)

File: src/Illuminate/Mail/Transport/MailgunTransport.php [hunk 1]
```diff
@@ -71,12 +79,10 @@ public function stop()
 	 */
 	public function send(Swift_Mime_Message $message, &$failedRecipients = null)
 	{
-		$client = $this->getHttpClient();
-
-		return $client->post($this->url, ['auth' => ['api', $this->key],
-			'body' => [
-				'to' => $this->getTo($message),
-				'message' => new PostFile('message', (string) $message),
+		return $this->client->post($this->url, ['auth' => ['api', $this->key],
```
Comment (jeremeamia) [inline src/Illuminate/Mail/Transport/MailgunTransport.php]: You can check `$client::VERSION` here and either use the v5 or v6 args to support both versions.

File: src/Illuminate/Mail/Transport/MandrillTransport.php [hunk 2]
```diff
@@ -54,10 +63,8 @@ public function stop()
 	 */
 	public function send(Swift_Mime_Message $message, &$failedRecipients = null)
 	{
-		$client = $this->getHttpClient();
-
-		return $client->post('https://mandrillapp.com/api/1.0/messages/send-raw.json', [
-			'body' => [
+		return $this->client->post('https://mandrillapp.com/api/1.0/messages/send-raw.json', [
```
Comment (jeremeamia) [inline src/Illuminate/Mail/Transport/MandrillTransport.php]: You can check `$client::VERSION` here and either use the v5 or v6 args to support both versions.

Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MandrillTransport.php]: right :)

File: src/Illuminate/Mail/Transport/MailgunTransport.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/MailgunTransport.php]: right :)

**Pull request merged**